image/distributionutil/pull.go

```diff
@@ -114,29 +117,50 @@ func (puller *ImagePuller) downloadLayer(ctx context.Context, named reference.Na
 
 // TODO make a manifest store do this job
 func (puller *ImagePuller) getRemoteManifest(context context.Context, named reference.Named) (schema2.Manifest, error) {
-	return puller.registry.ManifestV2(context, named.Repo(), named.Tag())
+	repo := puller.repository
+	ms, err := repo.Manifests(context)
+	if err != nil {
+		return schema2.Manifest{}, err
+	}
+
+	manifest, err := ms.Get(context, "", distribution.WithTagOption{Tag: named.Tag()})
+	if err != nil {
+		return schema2.Manifest{}, err
+	}
+
+	_, ok := manifest.(*schema2.DeserializedManifest)
+	if !ok {
+		return schema2.Manifest{}, fmt.Errorf("failed to parse manifest %s to DeserializedManifest", named.RepoTag())
+	}
+	return manifest.(*schema2.DeserializedManifest).Manifest, nil
 }
 
 // not docker image, get sealer image metadata
 func (puller *ImagePuller) getRemoteImageMetadata(context context.Context, named reference.Named, digest digest.Digest) (v1.Image, error) {
-	manifestImage, err := puller.registry.DownloadLayer(context, named.Repo(), digest)
+	repo := puller.repository
+	bs := repo.Blobs(context)
+	manifestImageBytes, err := bs.Get(context, digest)
 	if err != nil {
 		return v1.Image{}, err
 	}
 
-	decoder := json.NewDecoder(manifestImage)
 	img := v1.Image{}
-	return img, decoder.Decode(&img)
+	return img, json.Unmarshal(manifestImageBytes, &img)
 }
 
-func NewPuller(config Config) (Puller, error) {
-	reg, err := fetchRegistryClient(config.AuthInfo)
+func NewPuller(named reference.Named, config Config) (Puller, error) {
+	authInfo, err := utils.GetDockerAuthInfoFromDocker(named.Domain())
+	if err != nil {
+		logger.Warn("failed to get auth info, err: %s", err)
```

Comment:
If we need to return the error

Reply:
I guess we can ignore the error, just warn it, because authconfig is not vital in some scenes, like pulling from. public registry.